### ISSUE TITLE
Add error context from validator for use in withMessage

### ISSUE THREAD

**Issue by kkoster** [express-validator/express-validator#1194]
It would be nice if we could pass custom context associated with a validator  (mostly customValidator) to the custom message (withMessage) function. This is especially the case when a customValidator is applying complex rules that cannot be segregated into different validators. For example, within the body of a request I have an array of objects that have identifiers I need to validate against a database. Instead of validating them one at a time, I pass a vector of ids to an endpoint that validates existence an spits back the ids that successfully validated. The difference between what was submitted and what was received gives me the invalid ids. I would like to return the valid ids in a manner that the withMessage function can pick them up and incorporate them in the message.

Probably the most direct way to do this is to return an object with the ids (or whatever context I want to return) from the validator. A simple return on a synchronous validator or the content of the Promise.reject() on an asynchronous one. The validator framework could then pass that on to the withMessage function as an optional parameter (augmenting the Meta type for the withMessage function). That can then be inspected by the withMessage function for its use.

**Comment by gustavohenke** [comment]
Can you show a sample code of what you're doing/what you'd like express-validator to do?

**Comment by kkoster** [comment]
@gustavohenke for instance, within a validator I may need to access information from a database that has relevance to the error message I want to return:
```
.custom(async (dbId) => {
    dbObj = await dbUtility.get(dbId)
    if (dbObj.someProperty !== 'some value') {
        return Promise.reject(dbObj)
    }
})
.withMessage((dbId, { **context** /* new context object coming from the rejected promise */ }) => ({
        message: `The object ${dbId} has the value of ${context.someProperty} but should have 'some value'}`
    })
)
```
The withMessage could retrieve the database object all over again and provide the property value, but that puts additional overhead on the validation chain. And, if there are several of these in a chain, it could be come costly.

**Comment by gustavohenke** [comment]
If your custom validator rejects, the rejected value gets used as the message.
https://express-validator.github.io/docs/custom-error-messages#custom-validator-level

Does that help?

**Comment by gustavohenke** [comment]
> The withMessage could retrieve the database object all over again

Not really. At the moment `withMessage` has to be synchronous. It could be implemented, of course, but that wouldn't be DRY as you say 😄
